app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,70 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign_to = nil, cat = nil, objects = nil)
+    @assign_to = assign_to
+    @cat = cat
+    @objects = objects
+    build = false unless @assign_to
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    identifier = (object.name.presence || object.description)
+    if object.kind_of?(MiddlewareServer)
+      identifier += "-" + object.hostname
+    end
+
+    if @assign_to.ends_with?("-tags")
```

Comment:
Could you please move this logic to `x_get_tree_roots` where the `@assign_to` is evaluated one more time? This way we would spare an unnecessary double evaluation. Also this `override` is a last-resort solution and you should not use it when possible.